packages/modules/tesla/api.py

```diff
@@ -53,57 +51,38 @@ def request_soc_range(vehicle: int, token_file: str) -> Tuple[float, float]:
     return float(soc), range
 
 
-def __fetch_token(token_file):
-    token = {
-        "access_token": "",
-        "created_at": 0,
-        "expires_in": 0,
-        "refresh_token": ""
-    }
-
-    token, expiration = __load_token(token_file)
-    log.debug("No need to authenticate. Valid token already present in " + token_file)
+def validate_token(token: TeslaSocToken) -> TeslaSocToken:
+    expiration = token.created_at + token.expires_in
+    log.debug("No need to authenticate. Valid token already present.")
     if time.time() > expiration:
         log.debug("Access token expired. Refreshing token.")
-        token = __refresh_token(token_file, token)
+        token = __refresh_token(token)
     return token
 
 
-def __load_token(token_file: str) -> Tuple[Dict, int]:
-    with open(token_file, "r") as f:
-        token = json.load(f)
-        expiration = token["created_at"] + token["expires_in"]
-        return token, expiration
-
-
-def __refresh_token(token_file: str, token: Dict) -> Dict:
+def __refresh_token(token: TeslaSocToken) -> TeslaSocToken:
     headers = {"user-agent": UA, "x-tesla-user-agent": X_TESLA_USER_AGENT}
     payload = {
         "grant_type": "refresh_token",
         "client_id": "ownerapi",
-        "refresh_token": token["refresh_token"],
+        "refresh_token": token.refresh_token,
         "scope": "openid email offline_access",
     }
-    session = req.get_http_session()
-
-    resp = session.post("https://auth.tesla.com/oauth2/v3/token", headers=headers, json=payload, timeout=50)
-    resp_json = resp.json()
-    refresh_token = resp_json["refresh_token"]
+    resp = req.get_http_session().post("https://auth.tesla.com/oauth2/v3/token",
+                                       headers=headers,
+                                       json=payload,
+                                       timeout=50)
     log.debug("received refresh token")
-
-    # save our token
     resp_json = resp.json()
-    token["refresh_token"] = refresh_token
-    token["access_token"] = resp_json["access_token"]
-    token["created_at"] = resp_json["created_at"]
-    token["expires_in"] = resp_json["expires_in"]
+    token.refresh_token = resp_json["refresh_token"]
+    token.access_token = resp_json["access_token"]
+    token.created_at = resp_json["created_at"]
```

Comment:
Den Zeitstempel gibt es in der Antwort nicht.
```suggestion
    token.created_at = time.time()
```